waterloop-site/src/components/SustainableTech/Schedule/Agenda.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { createRef, useEffect, useState } from 'react';
+import styled from 'styled-components';
+
+import Content from '../../../static/copy/SustainableTech/Schedule.json';
+import Nodes from './Nodes';
+
+interface AgendaSection {
+  title: string;
+  subtitle?: string;
+  imagePath?: string;
+  text: string;
+  time: string;
+}
+
+const castData: AgendaSection[] = Content;
+const sectionBottomMargin = 50;
+
+const Container = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: flex-start;
+`;
+
+const ContentContainer = styled.div`
+  width: 50%;
+  margin-left: 50px;
+`;
+
+const SectionContainer = styled.div`
+  margin-bottom: ${sectionBottomMargin}px;
+`;
+
+const Title = styled.h3`
+  color: #80a773;
+  font-weight: bold;
+  font-size: 36px;
+`;
+
+const Subtitle = styled.p`
+  color: #3c6130;
+  font-weight: bold;
+`;
+
+const TextContainer = styled.div`
+  display: flex;
+  align-items: center;
+`;
+
+const SectionImage = styled.img.attrs((props) => ({
+  src: props.src,
+}))`
+  width: 75px;
+  height: 75px;
+  border-radius: 50%;
+  margin-right: 25px;
+`;
+
+const Text = styled.p``;
+
+const Agenda: React.FC = () => {
+  const [sectionRefs, setSectionRefs] = useState<any[]>([]);
+  const [sectionHeights, setSectionHeights] = useState<number[]>([]);
+
+  // Create refs for each schedule section, so that heights can be extracted
+  const getSectionRefs = () => {
+    setSectionRefs(
+      Array(castData.length)
+        .fill(null)
+        .map((_, i) => createRef()),
+    );
+  };
+
+  // Get the heights of each schedule section
+  const getSectionHeights = () => {
+    const heights = sectionRefs.map((ref) => ref.current?.offsetHeight);
+    setSectionHeights(heights);
+  };
+
+  useEffect(() => {
+    getSectionRefs();
+  }, []);
+
+  useEffect(() => {
+    getSectionHeights();
+  });
+
```

Comment:
Not sure how to go about fixing this but in the console the warning "Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render." shows up multiple times

Reply:
Hmm this is definitely a problem I need to fix. Everything I've tried thus far hasn't worked because it either messes up the node-edge heights or the infinite loop doesn't go away. I'll keep working for a solution.